### PR TITLE
Week 11: For Home

### PR DESCRIPTION
**Pull request by pvarna**

### CODE REVIEW

File: Week11-BigramLanguageModelUsingCounting/task01.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+def main():
+    names_file = open("../DATA/names.txt")
+    names = names_file.read().split('\n')
```
Comment (SimeonHristov99) [inline Week11-BigramLanguageModelUsingCounting/task01.py:3]: You have to also close the file. If you don't want to keep remembering that you could try the `with` clause which will automatically close the file, once the code execution exits it.

```python
    with open(os.path.join('DATA', 'names.txt'), 'r') as f:
        words = f.read().split()

    print(words[:10])
```

File: Week11-BigramLanguageModelUsingCounting/task11.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    names = names_file.read().split('\n')
+
+    bigram_matrix, vocabulary = create_bigram_matrix(names)
+    normalize_row_wise(bigram_matrix)
+
+    word = "simqeon"
+    prob = word_probability(word, bigram_matrix, vocabulary)
+    print(prob)
+
+    # Now the probability is not 0
```
Comment (SimeonHristov99) [inline Week11-BigramLanguageModelUsingCounting/task11.py:131]: Would be nice to document the actual value.

**Review by SimeonHristov99 — APPROVED** [review]
Perfect! Great job, Pesho!

File: Week11-BigramLanguageModelUsingCounting/task11.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    count = 0
+
+    for name in names:
+        bigrams = generate_bigrams(name)
+        for prev, next in bigrams:
+            i = char_to_idx[prev]
+            j = char_to_idx[next]
+            prob = bigram_matrix[i, j].item()
+
+            prob = max(prob, 0.00001)  # avoid log(0)
```
Comment (SimeonHristov99) [inline Week11-BigramLanguageModelUsingCounting/task11.py:98]: Would be nice to have this as a hyperparameter (regularization strength, lambda, something simialr), just like the learning rate would be for example.

**Review by SimeonHristov99 — APPROVED** [review]
